enhancements/network/user-defined-network-segmentation.md

```diff
@@ -413,6 +669,220 @@ guarantee uniqueness and eliminates the ability to falsify access.
+  ownerReferences:
+    - apiVersion: k8s.ovn.org/v1alpha1
+      blockOwnerDeletion: true
+      kind: ClusterUserDefinedNetwork
+      name: db-network
+      uid: f45efb13-9511-48c1-95d7-44ee17c949f4 
+spec:
+  config: >    
+      '{
+          "cniVersion":"0.3.1",
+          "excludeSubnets":"10.0.0.100/24",
+          "mtu":1500,   
+          "name":"shared-db-network",   <--- generated unique network-name
+          "netAttachDefName":"mynamespace/db-network",
+          "subnets":"10.0.0.0/24",
+          "topology":"layer2",
+          "type":"ovn-k8s-cni-overlay",
+          "role": "primary"
+      }'
+---
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: shared-db-network                 <--- same name as in other nameapces
+  namespace: theirnamespace
+  finalizers:
+  - k8s.ovn.org/user-defined-network-protection
+  ownerReferences:
+  - apiVersion: k8s.ovn.org/v1alpha1
+    blockOwnerDeletion: true
+    kind: ClusterUserDefinedNetwork
+    name: db-network
+    uid: f45efb13-9511-48c1-95d7-44ee17c949f4
+spec:
+  config: >
+    '{
+        "cniVersion":"0.3.1",
+        "excludeSubnets":"10.0.0.100/24",
+        "mtu":1500,   
+        "name":"shared-db-network",   <--- same name as in other namespaces
+        "netAttachDefName":"theirnamespace/db-network",
+        "subnets":"10.0.0.0/24",
+        "topology":"layer2",
+        "type":"ovn-k8s-cni-overlay",
+        "role": "primary"
+    }'
+```
+
+##### Validations
+The controller should validate the request CRD spec and verify:
+- CIDRs are valid.
+- MTU is valid, no more than 9000.
+- `Topology` is one of `Layer2`, `Layer3` or `Localnet`.
+- `Role` is one of `Primary` or `Secondary`.
+- `IPAMLifecycle` can be `Persistent`, and set only when topology is `Layer2` or `Localnet`. 
+- In case `Topology: Localnet`, `Role` cannot be `Primary`
+
+In addition, the following scenarios should be validated: 
+- The join subnet shall not overlap with the configured cluster default network join subnet.
+  If there is overlap, OVN-Kubernetes should report an error in the request CR status.
+
+- When primary network is requested (i.e.: `spec.role: Primary`)
+  - Verify no primary network exist at the target namespace (i.e.: no NAD with `"primaryNetwork": "true"` exist).
+    In case primary-network already exist, the request CR status should reflect network is not ready because primary network
+    already exist at the target namespace.
+
+- When CRD instance is deleted, ensure the network is not in use before continuing with deletion process (e.g.: remove finalizer).
+  - Check no pod using the CRD instance corresponding NAD.
+  - In case at least on pod using the network, update the status to reflect network cannot be deleted because its being used. 
+
+- When a managed NAD already exist at the target namespace:
+  - In case no owner-reference exist or owner-reference doesn't match the request CR object's UID, 
```

Comment:
Hmm but I thought you say earlier in the document:

Check that the desired NAD does not already exist. If not, create the NAD and return.
Otherwise, verify the existing NAD correspond to desired spec, if so return.